backupccl: create tables at start of restore

pbardea opened the pull request:
This change creates the tables that are to be restored at the start of
the RESTORE job. They are marked as ingesting and if the job succeeds
are marked public, and if not are marked to be DROPed.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/39698)
<!-- Reviewable:end -->

pbardea committed:
backupccl: create tables at start of restore

This change creates the tables that are to be restored at the start of
the RESTORE job. They are marked as OFFLINE and if the job succeeds
are marked public, and if not are marked to be DROPed.

Release note: None

dt reviewed (approved):
> Reviewed 2 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/39698)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt and @pbardea)

___
*[pkg/ccl/backupccl/restore.go, line 1792 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/39698#-LnEnLOp5DxjB49nRoac:-LnEnLOp5DxjB49nRoad:b-oragdz) ([raw file](https://github.com/cockroachdb/cockroach/blob/f69a6f3455f31ac4a46c3cddf8c03335369e1dfb/pkg/ccl/backupccl/restore.go#L1792)):*
> ```Go
> 		return err
> 	}
> 	b := txn.NewBatch()
> ```

should this be conditional on prepare complete?


<!-- Sent from Reviewable.io -->

pbardea reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/39698)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt)

___
*[pkg/ccl/backupccl/restore.go, line 1792 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/39698#-LnEnLOp5DxjB49nRoac:-LnF3ey64UOc_QAAXJmD:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/f69a6f3455f31ac4a46c3cddf8c03335369e1dfb/pkg/ccl/backupccl/restore.go#L1792)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

should this be conditional on prepare complete?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

dt reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/39698)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->
